core/flow/rule.go

```diff
@@ -102,6 +132,17 @@ type Rule struct {
 	// If the StatIntervalInMs user specifies can not reuse the global statistic of resource,
 	// 		sentinel will generate independent statistic structure for this rule.
 	StatIntervalInMs uint32 `json:"statIntervalInMs"`
+
+	// adaptive flow control alg
+	// limitation: SafeThreshold > RiskThreshold && HighWaterMark > LowWaterMark
+	// If the watermark is less than Rule.LowWaterMark, the threshold is Rule.SafeThreshold.
+	// If the watermark is greater than Rule.HighWaterMark, the threshold is Rule.RiskThreshold.
+	// Otherwise, the threshold is ((watermark - LowWaterMark)/(HighWaterMark - LowWaterMark)) *
+	//	(SafeThreshold - RiskThreshold) + RiskThreshold.
+	SafeThreshold int64 `json:"safeThreshold"`
```

Comment:
Currently, these fields are related to memory adaptation, but the field names do not give the related meaning. I don't have a better suggestion at the moment. We may need more discussion.